pinot-tools/src/main/java/org/apache/pinot/tools/perf/PerfBenchmarkDriver.java

```diff
@@ -237,18 +239,31 @@ private void startServer()
 
   private void startHelixResourceManager()
       throws Exception {
-    _helixResourceManager = new PinotHelixResourceManager(getControllerConf());
-    _helixResourceManager.start();
-
-    // Create broker tenant.
-    Tenant brokerTenant = new TenantBuilder(_brokerTenantName).setRole(TenantRole.BROKER).setTotalInstances(1).build();
-    _helixResourceManager.createBrokerTenant(brokerTenant);
-
-    // Create server tenant.
-    Tenant serverTenant =
-        new TenantBuilder(_serverTenantName).setRole(TenantRole.SERVER).setTotalInstances(1).setOfflineInstances(1)
-            .build();
-    _helixResourceManager.createServerTenant(serverTenant);
+    if (_conf.shouldStartController()) {
+      // helix resource manager is already available at this time if controller is started
+      _helixResourceManager = _controllerStarter.getHelixResourceManager();
+    } else {
+      // When starting server only, we need to change the controller port to avoid registering controller helix
+      // participant with the same host and port.
+      ControllerConf controllerConf = getControllerConf();
+      controllerConf.setControllerPort(Integer.toString(_conf.getControllerPort() + 1));
```

Comment:
In my opinion `_helixResourceManager` shouldn't even be instantiated here. Looking through the code, it only does the following steps:
* create tenants
* add table
* add segment

All these behaviors should be done by sending requests to controller.

Reply:
Good point. I already attempted that approach but we have some limits. We can easily change it for tenants and table creation. However, one issue is that we are taking different code path for segment upload. `addNewSegment()` is used for mostly tests and the function only updates `segmentZkMetadata`. We use this approach because extracted segment files are already available when starting a server in a perfBenchmarkRunner. If we want to use the controller API, we will have to tar segment files and upload to controller.

Reply:
I see. Thanks for the explanation! Can you add some comments on `_helixResourceManager` to explain a bit why it's needed here? I'm sure we're not the last one who had question on it.

Reply:
I added some comments. Can you check again?